pkg/operator/rules.go

```diff
@@ -257,7 +267,7 @@ type rulesValidator struct {
 }
 
 func (v *rulesValidator) ValidateCreate(ctx context.Context, o runtime.Object) error {
-	_, err := generateRules(o.(*monitoringv1.Rules), v.opts)
+	_, err := generateRules(o.(*monitoringv1.Rules), "test_project", "test_location", "test_cluster")
```

Comment:
Is this right?

Reply:
Yes. So it's hard to tell what's happening here, but `rulesValidator` [implements](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/pintohutch/opcfg-ext-labels/pkg/operator/operator.go#L361-L363) the [`admission.CustomValidator`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/pintohutch/opcfg-ext-labels/vendor/sigs.k8s.io/controller-runtime/pkg/webhook/admission/validator_custom.go#L30-L31) interface from the controller-runtime library. i.e. this method is executed as part of the admission webhooks for our `Rules` CRD. 

So when a new `Rule` is created, we essentially sanity-check it here with toy `project`, `location`, and `cluster` labels to ensure it can generate valid Prometheus rules. Thus, we only care if it returns and error, and we throw away the generated rules value.